src/epm_dynamic.rs

```diff
@@ -0,0 +1,213 @@
+}
+
+fn read_constant<'a, T: serde::Deserialize<'a>>(
+	api: &SubxtClient,
+	constant: EpmConstant,
+) -> Result<T, Error> {
+	let (epm_name, constant) = constant.to_parts();
+
+	let val = api
+		.constants()
+		.at(&subxt::dynamic::constant(epm_name, constant))
+		.map_err(|e| invalid_metadata_error(constant.to_string(), e))?;
+
+	scale_value::serde::from_value::<_, T>(val).map_err(|e| {
+		Error::InvalidMetadata(format!("Decoding `{}` failed {}", std::any::type_name::<T>(), e))
+	})
+}
+
+pub fn signed_solution<S: NposSolution + Encode + TypeInfo + 'static>(
+	solution: RawSolution<S>,
+) -> Result<DynamicTxPayload<'static>, Error> {
+	let scale_solution = to_scale_value(solution).map_err(|e| {
+		Error::DynamicTransaction(format!("Failed to decode `RawSolution`: {:?}", e))
+	})?;
+
+	Ok(subxt::dynamic::tx(EPM_PALLET_NAME, "submit", vec![scale_solution]))
+}
+
+pub fn unsigned_solution<S: NposSolution + Encode + TypeInfo + 'static>(
+	solution: RawSolution<S>,
+	witness: SolutionOrSnapshotSize,
+) -> Result<DynamicTxPayload<'static>, Error> {
+	let scale_solution = to_scale_value(solution)?;
+	let scale_witness = to_scale_value(witness)?;
+
+	Ok(subxt::dynamic::tx(EPM_PALLET_NAME, "submit_unsigned", vec![scale_solution, scale_witness]))
+}
+
+pub async fn signed_submission_at<S: NposSolution + Decode + TypeInfo + 'static>(
+	idx: u32,
+	at: Hash,
+	api: &SubxtClient,
+) -> Result<Option<SignedSubmission<S>>, Error> {
+	let scale_idx = Value::u128(idx as u128);
+	let addr = subxt::dynamic::storage(EPM_PALLET_NAME, "SignedSubmissionsMap", vec![scale_idx]);
+
+	match api.storage().fetch(&addr, Some(at)).await {
+		Ok(Some(val)) => {
+			let val = val.remove_context();
+			let v = encode_scale_value::<SignedSubmission<S>>(&val)?;
+			let submissions = Decode::decode(&mut v.as_ref())?;
+			Ok(Some(submissions))
+		},
+		Ok(None) => Ok(None),
+		Err(err) => Err(err.into()),
+	}
+}
+
+fn make_type<T: scale_info::TypeInfo + 'static>() -> (TypeId, PortableRegistry) {
+	let m = scale_info::MetaType::new::<T>();
+	let mut types = scale_info::Registry::new();
+	let id = types.register_type(&m);
+	let portable_registry: PortableRegistry = types.into();
+
+	(id.into(), portable_registry)
+}
+
+fn to_scale_value<T: scale_info::TypeInfo + 'static + Encode>(val: T) -> Result<Value, Error> {
+	let (ty_id, types) = make_type::<T>();
+
+	let bytes = val.encode();
+
+	decode_as_type(&mut bytes.as_ref(), ty_id, &types)
+		.map(|v| v.remove_context())
+		.map_err(|e| {
+			Error::DynamicTransaction(format!(
+				"Failed to decode {}: {:?}",
+				std::any::type_name::<T>(),
+				e
+			))
+		})
+}
+
+fn encode_scale_value<T: TypeInfo + 'static>(val: &Value) -> Result<Vec<u8>, Error> {
+	let (ty_id, types) = make_type::<T>();
+
+	let mut bytes = Vec::new();
+
+	encode_as_type(val, ty_id, &types, &mut bytes).map_err(|e| {
+		Error::DynamicTransaction(format!(
+			"Failed to encode {}: {:?}",
+			std::any::type_name::<T>(),
+			e
+		))
+	})?;
+	Ok(bytes)
+}
```

Comment:
If you wanted to pass this to some call (`api.tx().submit..`) as a `Value`, this would use the runtime metadata to encode properly I'd hope. But I'm guessing you ran into a problem or couldn't do that so I'm intrigued :)

Reply:
I had to use this because I wanted to read a storage value and then `codec::Decode::decode -> SignedSubmission` see `fn signed_submission_at` above.

The reason is that the `epm pallet` types doesn't implement serde, not sure if there is a more clever way to do it :)